### DISCUSSION TITLE
How can I achieve 'base > overlay > texture that only affects overlay'

### DISCUSSION THREAD

**Discussion by Zeioth** [ImageMagick/ImageMagick#5343] [Help]
I've managed to achieve the result I want: 

* A base image (shirt).
* A transparent overlay (tux).
* A transparent texture that affects only the overlay (fabric texture).

```
magick ./base.tiff \
  \( ./overlay.tiff "./texture.png" -composite ./overlay.tiff \) \
  -composite "./result.png"
```

![screenshot_2022-07-21_02-45-19_063123025](https://user-images.githubusercontent.com/3357792/180106103-573cb203-2798-4197-9535-769773387eda.png)

The problem come when I try to move the result with -geometry like

```
magick ./base.tiff \
  \( ./overlay.tiff "./texture.png" -composite ./overlay.tiff \) \
  -geometry +30+30 -composite "./result.png"
```

![screenshot_2022-07-21_02-46-33_076679784](https://user-images.githubusercontent.com/3357792/180106217-776d36cc-231b-4bd7-869f-d4ccad8b9e1e.png)

Only part of the result move. Any idea why is this and how can I fix it? Thank you. 

I upload the files in case you wanna tweak. [test.zip](https://github.com/ImageMagick/ImageMagick/files/9155336/test.zip)

Version: Version: ImageMagick 7.1.0-43 Q16-HDRI x86_64 20237

**Comment by snibgo** [comment]
Why are you reading overlay.tiff twice? I don't think you want to.

**Reply by Zeioth** [reply]
I'm not sure why it achieves the result I look for. But it does.

**Comment by Zeioth** [comment]
I've managed to do it in two commands. I understand there is no way to do it in only one command (I'm curious about understanding magick syntax).

    magick ./overlay.tiff ./texture.png -composite ./overlay.tiff -compose copy_alpha -composite ./texturized-overlay.tiff && magick ./base.tiff ./texturized-overlay.tiff -geometry +715+305 -composite ./result.jpg

**Reply by fmw42** [reply]
When I run thus command, I see no evidence of the texture.tiff image. It looks simply like you are overlaying the overlay.tiff over the background.tiff.

What result do you actually want? Is your command correct?

**Comment by snibgo** [comment]
> I'm not sure why it achieves the result I look for. But it does.

> Only part of the result move. Any idea why is this and how can I fix it?

It doesn't do what you want when you have `-geometry`. The reason is that your second `-composite` has three images for its input. The third image is `overlay.tiff`, and that is used as a mask, and that result isn't what you want. I suggest you remove the third image, so there is no mask.
